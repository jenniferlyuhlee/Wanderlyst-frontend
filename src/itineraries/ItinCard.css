.ItinCard-body{
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.bi-heart, .bi-heart-fill{
    font-size: 2.5rem;
    cursor: pointer;   /* Indicate that it's clickable */
  }

.bi-heart-fill{
    color: red;
}