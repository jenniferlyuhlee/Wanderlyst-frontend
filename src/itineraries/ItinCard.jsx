import React from "react";
import { Link } from "react-router-dom";

function ItinCard({itinerary}){
    return(
        <Link to={`/itineraries/${itinerary.id}`}>
            <div>
                <h4>{itinerary.title}</h4>
                <p>{itinerary.city}, {itinerary.country}</p>
                <p>Duration: {itinerary.duration} days</p>
                <small>By: {itinerary.username}</small>
            </div>
        </Link>
    )
}

export default ItinCard;