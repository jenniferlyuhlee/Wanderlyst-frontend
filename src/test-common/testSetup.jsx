import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };