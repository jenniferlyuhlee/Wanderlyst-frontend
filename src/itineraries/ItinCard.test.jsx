import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter} from "react-router-dom";
import ItinCard from "./ItinCard";
import { expect, describe, test} from "vitest";

const testItin = {
    id: 1,
    username: "testUser",
    title: "testItin",
    city: "testCity",
    country: "testCountry",
    duration: 1, 
    description: "Itin for testing."
}


describe("ItinCard component", () => {
    let mockHasLikedItin;
    let mockToggleLike;
  
    // Mocking the hasLikedItin and toggleLike methods
    beforeEach(() => {
      mockHasLikedItin = vi.fn().mockReturnValue(false);
      mockToggleLike = vi.fn();
    });

    test("renders without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider hasLikedItin={mockHasLikedItin} 
                              toggleLike={mockToggleLike}>
                    <ItinCard itinerary={testItin}/>
                </UserProvider>
            </MemoryRouter>)
    });

    test("matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <UserProvider hasLikedItin={mockHasLikedItin} 
                            toggleLike={mockToggleLike}>
                    <ItinCard itinerary={testItin}/>
                </UserProvider>
            </MemoryRouter>)

        expect(asFragment()).toMatchSnapshot();
    });

    test("toggleLike is called when heart clicked", async () => {
        const {container} = render(
            <MemoryRouter>
                <UserProvider hasLikedItin={mockHasLikedItin} 
                            toggleLike={mockToggleLike}>
                    <ItinCard itinerary={testItin}/>
                </UserProvider>
            </MemoryRouter>)

        // "like" itinerary
        const likeBtn = container.querySelector(".bi-heart");
        fireEvent.click(likeBtn);

        // check that toggleLike is called when clicked
        await waitFor(() => expect(mockToggleLike).toHaveBeenCalled());
        // check that liked heart displays
        expect(container.querySelector(".bi-heart-fill")).toBeInTheDocument();
    });
});