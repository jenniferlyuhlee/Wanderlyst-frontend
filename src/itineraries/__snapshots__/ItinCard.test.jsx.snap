// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItinCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ItinCard color-card card my-3 p-4"
  >
    <a
      class="ItinCard-link"
      href="/itineraries/1"
    >
      <h4
        class="card-title"
      >
        testItin
      </h4>
    </a>
    <hr
      class="m-0"
    />
    <div
      class="ItinCard-body mt-1"
    >
      <div
        class="ItinCard-info"
      >
        <small
          class="card-subtitle m-0"
        >
          testCity, testCountry
        </small>
        <br />
        <small>
          Duration: 1 days
        </small>
        <p>
          <a
            href="/users/profile/testUser"
          >
            <b>
              testUser 
            </b>
          </a>
          Itin for testing.
        </p>
      </div>
      <div>
        <i
          class="bi bi-heart"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
